### ISSUE TITLE
House edge value should not change with player bet spread

### ISSUE THREAD

**Issue by mhluska** [mhluska/blackjack-simulator#43]
Currently the house edge is calculated by looking at amount wagered vs amount earned which of course will change based on the player decisions. One option here is to first run the simulator with flat betting and perfect basic strategy to get an accurate house edge. The other is to just estimate the house edge based on rules (e.g. double after split adds 0.x% etc).

**Comment by danielchines** [comment]
I agree it'd be cool to have the non-counting house edge alongside the $/hr edge. Unfortunately I don't think the second option will work, I tried simulating the rule change of H17 and S17 when you're flat betting vs with a giant bet spread and the effect of H17/S17 was very between the two. (Side note: Aren't the first two numbers too low? I feel like the house edge shouldn't be negative when you're flat betting and only playing basic strategy...)

![Screen Shot 2022-07-09 at 12 42 54 PM](https://user-images.githubusercontent.com/25040262/178115036-e65b02af-a298-4640-bf0a-f9d34131caa4.png)

The more accurate number would come from a separate simulation with flat betting, as annoying as that sounds. Or maybe hard-coded values could work if you were to "memoize" the simulated house edge for every combination of house rules, but would be a lot of overhead if there are changes to how the game is coded.

**Comment by mhluska** [comment]
Sorry for the 4 year response time on this.

You're right that the house edge shouldn't be negative with flat betting and basic strategy. The issue is that the simulation defaults to I18+Fab4 deviations, not pure basic strategy. So the reported "House Edge" is really the player's realized edge including count-based decisions and bet spread, not the pure mathematical house edge of the table rules.

Your observation about the H17/S17 difference shrinking with a spread (0.21% flat vs 0.03% spread) also makes sense. At high counts where you bet big, the deviation plays dominate the H17/S17 rule difference, so it gets diluted in the weighted average.

The fix here is really just a labeling issue. The value being calculated (amountEarned / amountWagered) is useful, it just represents "player edge given this strategy and spread" rather than "house edge of the table rules." A separate issue is that amountWagered doesn't account for doubles and splits, which also skews the number.

Not going to change this for now since the current value is arguably more useful for card counters than the pure house edge (which is a well-known lookup value). But worth renaming the label at some point.
